client/src/GameState.ts

```diff
@@ -95,6 +106,32 @@ export class GameState {
                 this.emitAndPlaceCurrentTetromino();
             }
         });
+        this.socket.on("playerTrade", (playerId, _, otherTradeState) => {
+            console.log("Request received");
+            if (otherTradeState == TradeState.Offered) {
+                this.tradeState = TradeState.Pending;
+                this.tradingPlayerId = playerId;
+            } else if (otherTradeState == TradeState.Accepted) {
+                this.tradeState = TradeState.NoTrade;
+                this.tradingPlayerId = null;
+            }
+        });
+        this.socket.on("sendTradePiece", (tetrominoType) => {
+            this.currentTetromino.swapPiece(tetrominoType);
```

Comment:
I'm thinking that the `swapPiece` method should be removed all together and this should be replaced with just a re-instantiation of the current tetromino. Not only is `swapPiece` duplicate code from the Tetromino constructor, but it has functional issues due to pre-existing state that isn't present when constructing. For example, in the case of trading with the same type of tetromino, the `setType` would return early and thus the monomino positions and rotations would be out of sync from the Tetromino position and rotation.

We would need to destroy the existing monomino sprites before reassigning `this.tetromino` though, so Tetromino should get a `destroy()` method:
```ts
destroy() {
        if (this.monominoes) {
            this.monominoes.forEach((monomino) => monomino.destroy());
        }
}
```
and `Tetromino.setType` should be updated to call it. Then here it would be something like
```ts
        this.socket.on("sendTradePiece", (tetrominoType) => {
            this.currentTetromino.destroy();
            this.currentTetromino = new Tetromino(tetrominoType, this.playerId);
            // ...
```